### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by adria0**
Adding some checks:

- `rust` just compiles and runs the tests
- `audit` to check if dependencies are safe
- `clippy` do not allow commits if clippy generates warnings
- `coverage`, runs coverage tests and reports for commit deltas. Note that is required to get the authorization `CODECOV_TOKEN` token from codecov.io. Please, note that the configuration file `./codecov.yml` is used with relaxed coverage deltas to allow painless refactors. RUSTFLAGS are quite weird but are the recommended ones to run the coverage.
- `rustfmt` does not allow commits if code has not been properly formatted

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - uses: actions/checkout@v1
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --no-fail-fast
+        env:
```
Comment (vorot93) [inline .github/workflows/coverage.yml]: why is this necessary?

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+name: Code-coverage
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
```
Comment (vorot93) [inline .github/workflows/coverage.yml]: nightly?

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+coverage:
+  status:
+    project:
+      default:
+        treshold: 2%
```
Comment (vorot93) [inline codecov.yml]: what does it do?

File: .github/workflows/clippy.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+name: Clippy
+jobs:
+  clippy_check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - run: rustup component add clippy
+      - uses: actions-rs/clippy-check@v1
+        with:
```
Comment (vorot93) [inline .github/workflows/clippy.yaml]: why is this necessary?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# ethereum-forkid
+
+![Rust](https://github.com/vorot93/ethereum-forkid/workflows/Rust/badge.svg)
+![Clippy](https://github.com/vorot93/ethereum-forkid/workflows/Clippy/badge.svg)
+![Audit](https://github.com/vorot93/ethereum-forkid/workflows/Audit/badge.svg)
+![Rustfmt](https://github.com/vorot93/ethereum-forkid/workflows/Rustfmt/badge.svg)
```
Comment (vorot93) [inline README.md]: Can we have a single badge for all pipelines? If not, it should be just one file, for the simple reason that we will only merge PRs that pass the *whole* pipeline, not just parts of it.

File: codecov.yml (hunk 3 above)
Comment (adria0) [reply codecov.yml]: By default, the delta for LOC coverage is 0%, so any minimal refactor can break the CI, I checked in other projects, and a 2% is enough to allow minimal code changes.

Comment (vorot93) [reply codecov.yml]: I don't think we need *any* minimal threshold per my comment below. Should it be 'threshold: 100%' for such?

File: .github/workflows/clippy.yaml (hunk 4 above)
Comment (adria0) [reply .github/workflows/clippy.yaml]: If you are talking about the `secrets.GITHUB_TOKEN` ?

Comment (vorot93) [reply .github/workflows/clippy.yaml]: yes, this is not necessary per
https://github.com/actions-rs/meta/blob/master/recipes/quickstart.md

Comment (adria0) [reply .github/workflows/clippy.yaml]: Ok, I was following the https://github.com/actions-rs/clippy-check doc, I'll check to do it without the token

Comment (vorot93) [reply .github/workflows/clippy.yaml]: I'm generally wary of setting up tokens unless absolutely necessary for maintenance and security reasons.

File: .github/workflows/coverage.yml (hunk 2 above)
Comment (adria0) [reply .github/workflows/coverage.yml]: I've copied it from some project, so at the time nighty was required to run the coverage. Let's me check if this is still required.

Comment (vorot93) [reply .github/workflows/coverage.yml]: This is because of nightly flags in `RUSTFLAGS` below. So the real question is why those are necessary (and if not, it should work on stable).

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (adria0) [reply .github/workflows/coverage.yml]: Just to get information about coverage even if tests fails. Is for free.

File: codecov.yml (hunk 3 above)
Comment (adria0) [reply codecov.yml]: In general, I think is a good idea, if someone writes new code, he is going to be _signaled_ about adding new tests, and makes an extra filter to go forward to the reviewing process. But I am ok if you think that we need to remove it.

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (vorot93) [reply .github/workflows/coverage.yml]: But if we don't enforce code coverage limit then it shouldn't fail at all, right?

File: README.md (hunk 5 above)
Comment (adria0) [reply README.md]: I do not see this option in github. Is not requiered to put all the checks in the same file, github only allows merging if all checks are ok (you can force it if you're a repo admin, btw).

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (adria0) [reply .github/workflows/coverage.yml]: I suppose, @vorot93, but I've not checked it.

File: .github/workflows/coverage.yml (hunk 2 above)
Comment (adria0) [reply .github/workflows/coverage.yml]: See https://github.com/rust-lang/rust/issues/42524, it seems that the issue is stuck.

File: .github/workflows/clippy.yaml (hunk 4 above)
Comment (adria0) [reply .github/workflows/clippy.yaml]: Fixed

File: .github/workflows/coverage.yml (hunk 2 above)
Comment (vorot93) [reply .github/workflows/coverage.yml]: If so, I think we should remove the coverage for now. Nightly flags can change at any time and we don't want CI breakage for external reasons.

Comment (adria0) [reply .github/workflows/coverage.yml]: There's always the option to create a branch with this github action when we want to check the coverage.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+![Code Coverage](https://github.com/vorot93/ethereum-forkid/workflows/Code-coverage/badge.svg)
```
Comment (vorot93) [inline README.md]: This badge should go too.

Comment (vorot93) [inline README.md]: This badge should go too.
